nipype/interfaces/ants/preprocess.py

```diff
@@ -0,0 +1,358 @@
+        metric_str = ("-m {metric_type}[{fixed_image},{moving_image},"
+                      "{metric_weight},{radius_or_bins},{sampling_strategy},"
+                      "{sampling_percentage}]")
+        format_args = {}
+        metric_args = ["metric_type", "fixed_image", "moving_image",
+                       "metric_weight", "radius_or_bins", "sampling_strategy",
+                       "sampling_percentage"]
+        for metric_arg in metric_args:
+            if isdefined(getattr(self.inputs, metric_arg)):
+                format_args[metric_arg] = getattr(self.inputs, metric_arg)
+        return metric_str.format(**format_args)
+
+    def _format_transform(self):
+        transform_str = "-t {}[{}]"
+        if (isdefined(self.inputs.transformation_model)
+                and isdefined(self.inputs.gradient_step_length)):
+            return transform_str.format(self.inputs.transformation_model,
+                                        self.inputs.gradient_step_length)
+        raise ValueError("Unable to format transformation_model argument")
+
+    def _format_output(self):
+        if (isdefined(self.inputs.output_transform_prefix)
+                and isdefined(self.inputs.output_warped_image)
+                and isdefined(self.inputs.output_average_image)):
+            return "-o [{},{},{}]".format(
+                self.inputs.output_transform_prefix,
+                self.inputs.output_warped_image,
+                self.inputs.output_average_image
+            )
+        elif isdefined(self.inputs.output_average_image):
+            return "-o {}".format(self.inputs.output_average_image)
+        else:
+            raise ValueError("Unable to format output due to lack of inputs.")
+
+    def _list_outputs(self):
+        outputs = self._outputs().get()
+        if isdefined(self.inputs.output_average_image):
+            outputs['average_image'] = (
+                os.path.abspath(self.inputs.output_average_image)
+            )
+        if isdefined(self.inputs.output_warped_image):
+            outputs['warped_image'] = (
+                os.path.abspath(self.inputs.output_warped_image)
+            )
+        if isdefined(self.inputs.output_transform_prefix):
+            fname = '{}MOCOparams.csv'.format(
+                self.inputs.output_transform_prefix
+            )
+            outputs['composite_transform'] = os.path.abspath(fname)
+        if (isdefined(self.inputs.write_displacement) and
+                isdefined(self.inputs.output_transform_prefix) and
+                self.inputs.write_displacement is True):
+            fname = '{}Warp.nii.gz'.format(self.inputs.output_transform_prefix)
+            outputs['displacement_field'] = os.path.abspath(fname)
+        return outputs
+
+class Matrix2FSLParamsInputSpec(BaseInterfaceInputSpec):
+    matrix = File(
+        exists=True,
+        desc='Motion crrection matrices to be converted into FSL style motion parameters',
+        mandatory=True
+    )
```

Comment:
@effigies I think the `name_source` only works for command line interfaces. (I remember working on this myself, but I don't recall having it merged).